### ISSUE TITLE
It looks like all of library-name dependencies have not built yet

### ISSUE THREAD

**Issue by B1Z1** [nrwl/nx#12502]
Hi, sometimes on our CI NX throws an error. If you need more information from us, I can try to send you

## Current Behavior
NX Randomly throws an error on clean CI instance

## Expected Behavior
There are no random errors 

## Steps to Reproduce
I don't know how to prepare repro steps. Small repository works well, but here's 300+ libraries and it's difficult to prepare repro steps

### Failure Logs
[2022-10-11T07:27:09.724Z] > nx run ppm-report:build:production
[2022-10-11T07:27:09.724Z] 
[2022-10-11T07:27:09.724Z] It looks like all of ppm-report's dependencies have not been built yet:
[2022-10-11T07:27:09.724Z] - ppm-box-box-application
[2022-10-11T07:27:09.724Z] - ppm-box-box-application-api
[2022-10-11T07:27:09.724Z] - ppm-box-box-application-app
[2022-10-11T07:27:09.724Z] - ppm-box-box-application-ui-switcher
[2022-10-11T07:27:09.724Z] - ppm-box-box-application-ui-module-picker
[2022-10-11T07:27:09.724Z] - ppm-box-box-application-ui-utils
[2022-10-11T07:27:09.724Z] - ppm-box-box-application-ui-command-palette
[2022-10-11T07:27:09.724Z] - ppm-box-box-application-ui-core
[2022-10-11T07:27:09.724Z] - ppm-box-box-application-ui-resync-modal
[2022-10-11T07:27:09.724Z] - ppm-box-box-application-ui-tutorial-scenario
[2022-10-11T07:27:09.724Z] - ppm-box-box-application-ui-module-icons
[2022-10-11T07:27:09.724Z] - ppm-box-box-application-ui-shareable
[2022-10-11T07:27:09.724Z] - ppm-box-box-application-ui-wizard
[2022-10-11T07:27:09.724Z] - ppm-boxarea-boxarea-task-boxarea-task-infrastructure
[2022-10-11T07:27:09.724Z] - ppm-global-technical-info
[2022-10-11T07:27:09.724Z] 

[2022-10-11T07:27:09.724Z] You might be missing a "targetDefaults" configuration in your root nx.json (https://nx.dev/reference/project-configuration#target-defaults),

[2022-10-11T07:27:09.724Z] or "dependsOn" configured in ppm-report's project.json (https://nx.dev/reference/project-configuration#dependson)

### Environment
   Node : 16.16.0
   OS   : darwin arm64
   yarn : 1.22.4

   nx : 14.7.13
   @nrwl/angular : 14.7.13
   @nrwl/cypress : 14.7.13
   @nrwl/detox : Not Found
   @nrwl/devkit : 14.7.13
   @nrwl/eslint-plugin-nx : 14.7.13
   @nrwl/expo : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : 14.7.13
   @nrwl/js : 14.7.13
   @nrwl/linter : 14.7.13
   @nrwl/nest : 14.7.13
   @nrwl/next : Not Found
   @nrwl/node : 14.7.13
   @nrwl/nx-cloud : 14.6.2
   @nrwl/nx-plugin : Not Found
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : 14.7.13
   @nrwl/web : 14.7.13
   @nrwl/workspace : 14.7.13
   typescript : 4.7.4
   ---------------------------------------
   Local workspace plugins:
   ---------------------------------------
   Community plugins:
   	 @ngneat/transloco: 4.1.1
   	 @ngrx/effects: 12.5.0
   	 @ngrx/entity: 12.4.0
   	 @ngrx/store: 12.5.0
   	 @ngrx/store-devtools: 12.5.0
   	 @rx-angular/cdk: 1.0.0-rc.1
   	 @rx-angular/state: 1.6.0
   	 @rx-angular/template: 1.0.0-beta.32
   	 @storybook/angular: 6.5.12
   	 @testing-library/angular: 11.0.4

**Comment by FrozenPandaz** [comment]
Sorry, it's really hard to identify the issue here.

What kind of application is it? NestJS? Angular?
Do those libraries have a `build` target?
Do you have the following config in `nx.json`?

```
targetDefaults: {
  build: {
    dependsOn: ['^build'],
    ...
  }
}
```

A repo with this issue would really help. :pray:

**Comment by B1Z1** [comment]
Hi! We are using Angular as our main application. Every library is buildable.

This is our target defaults:

  "targetDefaults": {
    "build": {
      "dependsOn": ["^build"]
    }
  }

Every library has build target. 

Also, it's difficult to prepare a repo with the issue, because we are using NX since April and it's the first time when we got an error like this. It started throwing error, when we migrated from 14.1.9 -> 14.7.13
